cmd/ci-operator/main.go

```diff
@@ -2060,6 +2066,20 @@ func (o *options) loadConfig(info *api.Metadata) (*api.ReleaseBuildConfiguration
 	return &configSpec, nil
 }
 
+func resolveNodeArchitectures(ctx context.Context, client coreclientset.NodeInterface) ([]string, error) {
+	ret := sets.NewString()
+	nodeList, err := client.List(ctx, meta.ListOptions{})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine the node architectures: %w", err)
+	}
+
+	for _, node := range nodeList.Items {
+		ret.Insert(node.Status.NodeInfo.Architecture)
```

Comment:
It's safe to grab the architecture from the status. https://redhat-internal.slack.com/archives/CK1AE4ZCK/p1678100046793209